git-subhistory.sh

```diff
@@ -267,18 +273,27 @@ subhistory_assimilate () {
 		revs_to_rewrite=ASSIMILATE_HEAD
 	fi
 
-	git filter-branch \
+        commit_count=$(git rev-list --count $revs_to_rewrite)
+
+        if test $commit_count -ne 0
+        then
+	    git filter-branch \
 		--original subhistory-tmp/filter-branch-backup \
 		--parent-filter "$parent_filter" \
 		--index-filter "$index_filter"  \
 		-- $revs_to_rewrite \
-	2>&1 | say_stdin || exit $?
-
-	say
-	say "Assimilated $assimilatee into $(
-		git symbolic-ref --short HEAD -q \
-		|| echo "detached HEAD ($(git rev-parse --short HEAD))"
-	) under $path_to_sub as ASSIMILATE_HEAD"
+	        2>&1 | say_stdin || exit $?
+
+	    say
+	    say "Assimilated $assimilatee into $(
+		    git symbolic-ref --short HEAD -q \
+		    || echo "detached HEAD ($(git rev-parse --short HEAD))"
+	    ) under $path_to_sub as ASSIMILATE_HEAD"
+        else
+            say
+            say "Already completely assimilated - nothing to do"
+	    git update-ref ASSIMILATE_HEAD HEAD || exit $?
```

Comment:
Without this, ASSIMILATE_HEAD points to non-assimilated commits (IE the trees are from the sub-project, not under `$path_to_sub`).  By setting ASSIMILATE_HEAD to HEAD, the subsequent `git merge ASSIMILATE_HEAD` will do the right thing (IE nothing).